### DISCUSSION TITLE
Performance Issues finally mostly fixed with the pre-release of 1.3.0

### DISCUSSION THREAD

**Discussion by theAkito** [Dygmalab/Bazecor#463] [General]
Been waiting really long for this app to finally become usable. It was sluggish as hell. Pretty much unusable in any scenario close to real life usage.

Whatever made that happen, this was needed a ton. While [this](https://github.com/Dygmalab/Bazecor/discussions/409#discussioncomment-5408669) explanation for why a ridiculous slowpoke, like JavaScript, was chosen, seems reasonable, I'm still pretty sure, that a major reason performance might be bad is because of the use of resource hogs, like Electron etc.

**Comment by javierguzman** [comment]
Hello @theAkito thanks for letting us know. Indeed Electron is not the best for making apps with good performance but it allows us to release on Windows, MacOS and Linux in one shot. We do not have manpower to create native apps for every platform but we have also taken a look at other technologies that allow multiplatform (e.g. Tauri) but that would require a lot of rewrite to be honest.

In regards to the performance improvement is due to a major upgrade we have done to Bazecor internals. Basically we have move to electron forge to have a better build system and also we have upgraded all libraries used.

By the way, what OS did you use? It is just curiosity.

Again thanks for your words

**Reply by theAkito** [reply]
Thank you for your interest in what I mentioned. 🙂

Using Windows 10, but when I get to use this software seriously, I'll also use it on Linux, too.

Just to clarify, I did not try to blame or complain about it. I just wanted to mention, that I'm happy it's now usable & that I also agree with the fact, that JavaScript/Electron was chosen as the base & therefore is not performant. 😄

I did not know about Tauri yet, but as far as I understand its backend will always be based on Rust, no matter which language or framework you use for the frontend, as long as it ends up as CSS, HTML & JavaScript.
If this is the case, you are on the right track!
Currently, anything regarding Rust is pretty much always a safe bet.

I wish you the best & hope more contributors will be contributing to this project!

If some Rust code were involved I might even contribute myself. 😁

**Reply by javierguzman** [reply]
No worries! I understood it was not a complain and it is was more a message to say you were happy with the update. 
 
 When you try on Linux, please, let us know, because it was a bit flaky when we tested it here. 
 
 The problem with Rust is the rewrite we would have to do and also the learning curve, though we would really like to move. Also, the community of Tauri is not very big yet as it is still early days. If we face serious problems then not many people will be able to help, I mean, even with Electron that has a bigger company we have faced problems that nobody has responded yet...
 
 Thank you for your words and wishes!
